addons/sys_rack/fnc_isRackHearable.sqf

```diff
@@ -30,4 +30,10 @@ if (_unit == vehicle _unit) exitWith {false};
 
 private _wiredIntercoms = [_rackId] call FUNC(getWiredIntercoms);
 
-("crew" in _wiredIntercoms && {_unit in ACRE_PLAYER_CREW_INTERCOM}) || ("passenger" in _wiredIntercoms && {_unit in ACRE_PLAYER_PASSENGER_INTERCOM})
+private _isHearable = false;
+{
+    _isHearable = [_vehicle, _unit, _forEachIndex] call EFUNC(sys_intercom,isInIntercom) && (_x ==  (_wiredIntercoms select _forEachIndex));
+    if (_isHearable) exitWith{};
+} forEach (_vehicle getVariable [QEGVAR(sys_intercom,intercomNames), []]);
+
+_isHearable
```

Comment:
Whole loop as:
```sqf
{
    if ([_vehicle, _unit, _forEachIndex] call EFUNC(sys_intercom,isInIntercom) && {_x ==  _wiredIntercoms select _forEachIndex}) exitWith {true};
} forEach (_vehicle getVariable [QEGVAR(sys_intercom,intercomNames), []]);

false
```
Maybe multi-line due to length:
```sqf
{
    if ([_vehicle, _unit, _forEachIndex] call EFUNC(sys_intercom,isInIntercom) &&
        {_x ==  _wiredIntercoms select _forEachIndex}
    ) exitWith {true};
} forEach (_vehicle getVariable [QEGVAR(sys_intercom,intercomNames), []]);

false
```

Reply:
I tested this to make sure. If you implement it like you do, it always return false since you exit the loop with true, but the function returns always false.

Reply:
You are correct, my bad. Funky `exitWith`.